geomstats/special_orthogonal_group.py

```diff
@@ -557,94 +557,303 @@ def matrix_from_quaternion(self, quaternion):
+                    cos_angle_y = gs.cos(angle_y[i])
+                    sin_angle_y = gs.sin(angle_y[i])
+                    cos_angle_z = gs.cos(angle_z[i])
+                    sin_angle_z = gs.sin(angle_z[i])
+
+                    column_1 = [cos_angle_x * cos_angle_y,
+                                cos_angle_y * sin_angle_x,
+                                - sin_angle_y]
+                    column_2 = [(cos_angle_x * sin_angle_y * sin_angle_z
+                                 - cos_angle_z * sin_angle_x),
+                                (cos_angle_x * cos_angle_z
+                                 + sin_angle_x * sin_angle_y * sin_angle_z),
+                                + cos_angle_y * sin_angle_z]
+                    column_3 = [(sin_angle_x * sin_angle_z
+                                 + cos_angle_x * cos_angle_z * sin_angle_y),
+                                (cos_angle_z * sin_angle_x * sin_angle_y
+                                 - cos_angle_x * sin_angle_z),
+                                cos_angle_y * cos_angle_z]
+
+                    rot_mat[i] = gs.vstack([column_1, column_2, column_3])
+
+        elif extrinsic_or_intrinsic == 'intrinsic':
+            if order == 'zyx':
+                rot_mat = self.matrix_from_tait_bryan_angles(
+                    tait_bryan_angles,
+                    extrinsic_or_intrinsic='extrinsic',
+                    order='xyz')
+
+            elif order == 'xyz':
+                rot_mat = self.matrix_from_tait_bryan_angles(
+                    tait_bryan_angles,
+                    extrinsic_or_intrinsic='extrinsic',
+                    order='zyx')
 
-        quaternion = gs.zeros((n_yaw_pitch_rolls, 4))
-
-        cos_half_angle = (cos_half_yaw * cos_half_pitch * cos_half_roll
-                          + sin_half_yaw * sin_half_pitch * sin_half_roll)
+        else:
+            raise ValueError('extrinsinc_or_intrinsic should be'
+                             ' \'extrinsic\' or \'intrinsic\'.')
 
-        quaternion[:, 0] = cos_half_angle
+        return rot_mat
 
-        quaternion[:, 1] = (cos_half_yaw * cos_half_pitch * sin_half_roll
-                            - sin_half_yaw * sin_half_pitch * cos_half_roll)
+    def tait_bryan_angles_from_matrix(self, rot_mat,
+                                      extrinsic_or_intrinsic='extrinsic',
+                                      order='zyx'):
+        """
+        Convert a rotation matrix rot_mat into the tait bryan angles,
+        [angle_x, angle_y, angle_z] in extrinsic (fixed) coordinate frame,
+        for the order zyx, i.e.:
+        rot_mat = X(angle_x).Y(angle_y).Z(angle_z)
+        where:
+        - X(angle_x) is a rotation of angle angle_x around axis x.
+        - Y(angle_y) is a rotation of angle angle_y around axis y.
+        - Z(angle_z) is a rotation of angle angle_z around axis z.
+        """
+        rot_mat = gs.to_ndarray(rot_mat, to_ndim=3)
+        quaternion = self.quaternion_from_matrix(rot_mat)
```

Comment:
You mean, rename all the conversion functions? I liked the _from_ syntax, because you have : quaternion - quaternion - matrix - matrix in this order when you read:        
`quaternion = self.quaternion_from_matrix(rot_mat)`